runtime/test_artifacts/generator.go

```diff
@@ -145,19 +146,26 @@ func NewAccounts(n int, privateKeys ...string) (accounts []modules.Account) {
 	return
 }
 
+// TODO: The current implementation of NewActors  will have overlapping `ServiceUrl` for different
+//       types of actors which needs to be fixed.
 func NewActors(actorType typesUtil.ActorType, n int) (actors []modules.Actor, privateKeys []string) {
 	for i := 0; i < n; i++ {
-		genericParam := fmt.Sprintf("node%d.consensus:8080", i+1)
+		genericParam := GetServiceUrl(i + 1)
 		if int32(actorType) == int32(types.ActorType_App) {
 			genericParam = DefaultMaxRelaysString
 		}
 		actor, pk := NewDefaultActor(int32(actorType), genericParam)
 		actors = append(actors, actor)
 		privateKeys = append(privateKeys, pk)
 	}
+
 	return
 }
 
+func GetServiceUrl(n int) string {
```

Comment:
TODO revert or DISCUSS how we are importing `generator.go` code in order to satisfy previously encapsulated tests. If we decide to have a shared `testing utils` package - that's fine, but the `generator` has a specific use case to create the build files and I don't think this is an appropriate import

Reply:
I fixed the encapsulation logic. Lmk if you want me to remove this change too though since it is a "fly by side change".

I have this comment in the `p2p/README` but might be worth discussing (or scoping into a specific piece of work):

```
- **Variable Coupling**:There is an implicit coupling between `validatorId`, `serviceUrl` and `genericParam` that requires understanding of the codebase. Reach out to @olshansk or @andrewnguyen22 for clarity on this.
```